lib/rsolr/error.rb

```diff
@@ -24,6 +26,18 @@ def to_s
 
     def parse_solr_error_response body
       begin
+        # Default JSON response, try to parse and retrieve error message
+        if response[:headers] && response[:headers]["content-type"].start_with?("application/json")
+          begin
+            parsed_body = JSON.parse(body)
+            info = parsed_body && parsed_body["error"] && parsed_body["error"]["msg"]
```

Comment:
i follow this pattern all the time, however isn't there a more rubyish way to avoid nils?

Reply:
In ruby 2.3 and above, perhaps:

    parsed_body.dig("error", "msg")

Although you still need to make sure parsed_body is not nil, so maybe

   parsed_body && parsed_body.dig("error", "msg")

I wasn't sure exactly what versions of ruby are supported by this gem, and went with matching the patterns used elsewhere in this file over being inconsistent, or changing more of the file than I needed to to make the change I needed. 

This is very old little-maintained code, where it is very hard to get code review on any changes, and it's unclear if there is anyone paying attention go big-picture concerns, so my goals are generally to be change as little code as possible to make the change I need, and to be as consistent style with existing code as possible.